cardano-node/src/Cardano/Node/Types.hs

```diff
@@ -274,6 +278,25 @@ data NodeHardForkProtocolConfiguration =
      }
   deriving (Eq, Show)
 
+-- | A newtype wrapper around 'UseLedgerAfter' which provides 'FromJSON' and
+-- 'ToJSON' instances.
+--
+-- 'UseLedgerAfter' is used to configure from which slot a p2p node can use on
+-- chain root peers.
+--
```

Comment:
If `UseLedgerAfter` is concerned with a p2p node using on chain root peers why not a name like `UseOnChainRootPeersAfter`? If this ends up in the node tracing then the `ToJSON` instance should be a bit more explanatory for users debugging the node.